### ISSUE TITLE
[macOS] Launching the Game on Steam pops no windows, yet the game never launches

### ISSUE THREAD

**Issue by KevinShengZK** [bluemagic123/tModLoader#3031]
### Version

Latest 1.4-stable

### OS

Mac

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

MacOS Monterey - M1 Chip

Whenever I click "Play" in the steam library, it (after steam's series of cloud syncing process) pops up nothing and then stops itself, as the "stop" button turns back to "play"
Then i found that it was actually the problem of dotnet. The directory /usr/Library/Application Support/Steam/steamapps/common/tModLoader/dotnet/6.0.0 was empty, and as in log file below, it tried to install dotnet but fails 
[Launch(1).log](https://github.com/tModLoader/tModLoader/files/9771044/Launch.1.log)
I then tried to install dotnet manually on dotnet.microsoft.com in this link: 
https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-6.0.0-macos-x64-binaries
I moved everything from the downloaded .tar file to the dotnet directory (.../dotnet/6.0.0) and then click "Play" in steam library, but got a pop up from mac OS as below
<img width="266" alt="Screen Shot 2022-10-13 at 13 30 00" src="https://user-images.githubusercontent.com/91671708/195509603-d810aa28-79c7-44ea-b89b-d79933dd4301.png">
and the Native log as below, matching to this problem:
[Natives.log](https://github.com/tModLoader/tModLoader/files/9771059/Natives.log)

So I still cannot play the game. No windows except the damaged application pop-up reveals when i click "Play", as short while after it turns to "Stop" it turns back to "Play".

### Log File

Both logs after i installed dotnet 6.0.0 Runtime from https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-6.0.0-macos-x64-binaries
and ran the game as below:

[Launch.log](https://github.com/tModLoader/tModLoader/files/9771077/Launch.log)
[Natives.log](https://github.com/tModLoader/tModLoader/files/9771078/Natives.log)

### Steps to reproduce

1. Install binary dotnet from https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-6.0.0-macos-x64-binaries 
2. After removing everything from the following directory, drag all contents from the downloaded extracted file into the ...usr/Library/Application Support/Steam/steamapps/common/tModLoader/dotnet/6.0.0 directory
3. Click tModLoader "Play" in Steam Library

*The Reason why i choose this specific version is that i find only installing this one does not create the "wrong dotnet version" error. Is it the right one? If the game should have downloaded dotnet automatically, then as in Descriptions, i could not download it somehow as it keeps failing

### Expected Behavior

a

### Actual Behavior

What I get is still no window pop out, and the "Play" button, after becoming "Stop" turns back to "Play" in a few seconds. Only this pop out:
<img width="266" alt="Screen Shot 2022-10-13 at 13 30 00" src="https://user-images.githubusercontent.com/91671708/195509603-d810aa28-79c7-44ea-b89b-d79933dd4301.png">

### Reproduction frequency

a

### Additional Information

a

**Comment by KevinShengZK** [comment]
When I trash the app as macOS told me to do that was said to be "damaged", it breaks the framework.
"It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '6.0.0' (x64) was not found.
  - No frameworks were found.

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=osx.11.1-x64"
from Natives.log

**Comment by KevinShengZK** [comment]
> When I trash the app as macOS told me to do that was said to be "damaged", it breaks the framework. "It was not possible to find any compatible framework version The framework 'Microsoft.NETCore.App', version '6.0.0' (x64) was not found.
> 
> * No frameworks were found.
> 
> You can resolve the problem by installing the specified framework and/or SDK.
> 
> The specified framework can be found at:
> 
> * https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=osx.11.1-x64"
>   from Natives.log

Installing the installer from the given link does not solve the problem. In fact it does not even change log file, like the launcher ignored it.

**Comment by KevinShengZK** [comment]
New Progress: it seemed like, that it was the problem of macOS files. in tModLoader/dotnet/6.0.0/shared, Microsoft.NETCore.App got converted to an application due to macOS, instead of what should be directory to acces slibrary from.
<img width="338" alt="image" src="https://user-images.githubusercontent.com/91671708/195594057-3e6b37c1-d06b-4238-90e5-d793794f9b9a.png">
Which executing it makes the system think that the bash code is running the application, which shouts out an error of this being a "damaged application", whom should have never been an application at all. The way macOS automatically package directories ended with .app to applications blocked tMod's access to dotnet libraries. Maybe changing the root will help? Or what can I do to fix this.

**Comment by KevinShengZK** [comment]
Or, is it, afterall i should install the other version of dotnet?

**Comment by KevinShengZK** [comment]
ITS FIXED BOIIIIII
THIS IS HOW:
I first copied all the directory from the broken Microsoft.NETCore.App.app to somewhere else, temporarily. Then, I removed the old Microsoft.NETCore.App.app, created a new directory, naming it "Microsoft.NETCore.App" and the directory automatically gets converted to a "damaged application" with name "Microsoft.NETCore.App.app" (which should look the exact same as previous one). Then paste all the contents from the old application to the new broken application, and try launch game, as it launches! (Actually in between I have also changed the path in LaunchUtils/InstallNetFramework.sh, and after seeing that the new broken application can be visited by bash scripts by trying to launch the game, i changed it back)
